tests/conftest.py

```diff
@@ -909,6 +911,42 @@ def create_mock_mqtt(*args, **kwargs) -> MqttMockHAClient:
         yield _setup_mqtt_entry
 
 
+@pytest.fixture
+def yaml_config() -> ConfigType | None:
+    """Fixture to override configuration.yaml using mock_yaml_config."""
+    return None
+
+
+@pytest.fixture
+def mock_yaml_config(
+    hass: HomeAssistant, tmp_path: Path, yaml_config: ConfigType
+) -> Generator[Path, None, None]:
```

Comment:
```suggestion
) -> Generator[ConfigType, None, None]:
```

Reply:
Oef

Reply:
Isn't this generator return a `Path`?

Reply:
Indeed - the yield statement was causing confusion.